stripe/src/main/java/com/stripe/android/paymentsheet/BaseAddCardFragment.kt

```diff
@@ -181,19 +182,24 @@ internal abstract class BaseAddCardFragment(
     }
 
     private fun updateSelection() {
-        sheetViewModel.updateSelection(
-            if (addCardViewModel.isCardValid) {
-                paymentMethodParams?.let { params ->
-                    PaymentSelection.New.Card(
-                        params,
-                        cardMultilineWidget.brand,
-                        shouldSavePaymentMethod = shouldSaveCard()
-                    )
-                }
-            } else {
-                null
+        val validCard = if (addCardViewModel.isCardValid) {
```

Comment:
Can we add update `PaymentSheetAddCardFragmentTest` for this new logic?

Reply:
Hmm, I like the idea. I just added a test to check the newCard is set.  However, verifying the fields are populated on start is a little tricky because newCard needs to be set in the ViewModel.  Any ideas?

Reply:
Tests added see PaymentSheetAddCardFragmentTest